python/simulation.py

```diff
@@ -1115,6 +1116,34 @@ def require_dimensions(self):
         if self.structure is None:
             mp.set_dimensions(self._infer_dimensions(self.k_point))
 
+    def get_estimated_memory_usage(self):
+        if self.fields is None:
+            self.collect_stats = True
+            self.init_sim()
+
+        if self.fragment_stats is None:
+            self.fragment_stats = self._compute_fragment_stats(self.structure.user_volume)
+
+        is_periodic = self.k_point and self.k_point != mp.Vector3(0, 0, 0)
+        E_realnums = self.fragment_stats.num_pixels_in_box * (2 if is_periodic else 1)
+        M_realnums = self.fragment_stats.num_pixels_in_box * (2 if is_periodic else 1)
```

Comment:
What is M?

Reply:
I guess it was my short hand for "magnetic." My notes from when we talked about this say "E, M, and D" are the minimum three things that are always present.